multiplayer/src/components/HostLobby.tsx

```diff
@@ -1,65 +1,64 @@
 import { useContext, useRef, useState } from "react";
 import { Button } from "react-common/components/controls/Button";
-import { Input } from "react-common/components/controls/Input";
+import { Link } from "react-common/components/controls/Link";
 import { startGameAsync } from "../epics";
 import { clearModal } from "../state/actions";
 import { AppStateContext } from "../state/AppStateContext";
+import { makeJoinLink, SHORT_LINK } from "../util";
+import CopyButton from "./CopyButton";
+import Loading from "./Loading";
 import PresenceBar from "./PresenceBar";
 
 export default function Render() {
     const { state, dispatch } = useContext(AppStateContext);
-    const [copySuccessful, setCopySuccessful] = useState(false);
-    const inputRef = useRef<HTMLInputElement>(null);
-    const inviteString = lf("Invite anyone to join your game instantly. Just send them a link!");
 
     const onStartGameClick = async () => {
         pxt.tickEvent("mp.hostlobby.startgame");
         dispatch(clearModal());
         await startGameAsync();
     };
 
-    const handleCopyClick = () => {
-        pxt.tickEvent("mp.hostlobby.copyjoinlink");
-        if (pxt.BrowserUtils.isIpcRenderer()) {
-            if (inputRef.current) {
-                setCopySuccessful(
-                    pxt.BrowserUtils.legacyCopyText(inputRef.current)
-                );
-            }
-        } else {
-            navigator.clipboard.writeText(joinLink);
-            setCopySuccessful(true);
-        }
-    };
+    const joinCode = state.gameState?.joinCode;
+    if (!joinCode) {
+        return <Loading />;
+    }
 
-    const handleCopyBlur = () => {
-        setCopySuccessful(false);
-    };
+    // To get a link in the middle of the invite string, we actually preserve the {0} and insert the link manually as an html element later.
+    const inviteString = lf("Go to {0} and enter code", "{0}");
+    const inviteStringSegments = inviteString.split("{0}");
+    const shortLink = SHORT_LINK();
+
+    // Insert a space to make the join code easier to read.
+    const displayJoinCode = joinCode?.slice(0, 3) + " " + joinCode?.slice(3);
+    const joinDeepLink = makeJoinLink(joinCode);
 
-    const joinLink = `${state.gameState?.joinCode}`; // TODO multiplayer : create full link
     return (
         <div className="tw-flex tw-flex-col tw-gap-1 tw-items-center tw-justify-between tw-bg-white tw-py-[3rem] tw-px-[7rem] tw-shadow-lg tw-rounded-lg">
             <div className="tw-mt-3 tw-text-lg tw-text-center tw-text-neutral-700">
-                {inviteString}
+                {inviteStringSegments[0]}
```

Comment:
Assuming the number of elements in inviteStringSegments feels brittle to me. And assuming the link is in the middle segment isn't a safe assumption. I think you will need to iterate over the segments, and replace the one matching "{0}" with the link.

Reply:
Hmm, currently we split on {0} so there's no actual segment containing it. The only scenario where there wouldn't be 2 segments is if it was at the beginning or the end, so would it be better if we simply checked startsWith and endsWith {0} and reacted accordingly?

Reply:
I guess if {0} ended up in multiple places that'd also mess things up but I don't expect that would happen as a result of localization.

Reply:
split will leave an empty string at beginning / end if the the string starts with / ends with it, e.g. 
![image](https://user-images.githubusercontent.com/5615930/198417045-1d4ae407-3e22-46c6-9391-600caa691dd2.png)
I guess if the translation doubled up it would be an issue, but that would be pretty weird / we do this in a few other locations already

Reply:
I think I would split on whitespace and find the {0}. You could optimize by rejoining the contiguous word sequences before and after it. Do this in a useMemo.

Reply:
But there a ways of solving it.

Reply:
What's the benefit? Feels like unnecessary complexity to me.

Reply:
Leaving this as-is for now b/c time crunch, but we can follow up later to understand concerns and address as needed.